### DISCUSSION TITLE
Allow to open Pull Request in existing Codespace

### DISCUSSION THREAD

**Discussion by hediet** [community/community#41005] [Codespaces]
### Select Topic Area

Product Feedback

### Body

I am reviewing many pull requests and would like to do that in codespaces, so I can run/debug tests.

For that, I would like to open a pull request in an existing codespace (which has a different branch checked out before). Effectively, I want an existing codespace to open and checkout the PR I'm currently viewing in GitHub.

However, I don't see any option here:

![image](https://user-images.githubusercontent.com/2931520/206182070-c78681df-3bfb-4b01-9986-cc3c26f55fac.png)

I'd like to avoid creating a code space for every pull request I'm reviewing, as this creates unnecessary many codespaces and creating a codespace takes a long time (especially installing dependencies and running the watch task).

**Comment by joshaber** [comment]
Hi @hediet, we typically recommend creating a codespace per task so that they're independent. If Codespace creation time is too slow we recommend setting up prebuilds for the repository: https://docs.github.com/en/codespaces/prebuilding-your-codespaces/about-github-codespaces-prebuilds
